src/java/org/apache/cassandra/repair/SystemDistributedKeyspace.java

```diff
@@ -307,6 +320,42 @@ public static void setViewRemoved(String keyspaceName, String viewName)
         forceBlockingFlush(VIEW_BUILD_STATUS);
     }
 
+    /**
+     * Reads blacklisted partitions from system_distributed.blacklisted_partitions table
+     * @return
+     */
+    public static Set<BlacklistedPartition> getBlacklistedPartitions()
+    {
+        String query = "SELECT keyspace_name, columnfamily_name, partition_key FROM %s.%s";
+        UntypedResultSet results;
+        try
+        {
+            results = QueryProcessor.execute(format(query, SchemaConstants.DISTRIBUTED_KEYSPACE_NAME, BLACKLISTED_PARTITIONS),
```

Comment:
I am assuming `BLACKLISTED_PARTITIONS` table wont be too big that it eats up your heap. Do you have any restrictions on that?

Reply:
After further offline discussion, changed the cache to be bounded, in order to avoid possible OOM'ing of C*